test/integration/controlplane/transformation/transformation_testcase.go

```diff
@@ -161,6 +216,23 @@ func (e *transformTest) getETCDPath() string {
 	return fmt.Sprintf("/%s/secrets/%s/%s", e.storageConfig.Prefix, e.ns.Name, e.secret.Name)
 }
 
+func (e *transformTest) getETCDPathForResource(group, resource, name, namespaceName string) string {
+	if e.secret != nil {
```

Comment:
it's pretty hard to follow for state inside `transformTest` to override params... can we make this a pure function that takes `storagePrefix, group, resource, name, namespace` params?

Reply:
also, what's the difference between getETCDPathForResource and getETCDPath? can we collapse to a single etcd path computation function?

Reply:
> also, what's the difference between getETCDPathForResource and getETCDPath? can we collapse to a single etcd path computation function?

I didn't want to touch getETCDPath to minimize changes on existing test files. If you prefer to combine them, then I will go ahead and update kms_transformation_test.go and kmsv2_transformation_test.go. LMK!

Reply:
the ripples don't look *too* bad and I think leave the test package in a less confusing state (mostly just makes the paths that assume they are dealing with secrets explicitly say so)

I got a clean run with this diff:

```diff
diff --git a/test/integration/controlplane/transformation/kms_transformation_test.go b/test/integration/controlplane/transformation/kms_transformation_test.go
index 9773296ee29..3db46c22c3b 100644
--- a/test/integration/controlplane/transformation/kms_transformation_test.go
+++ b/test/integration/controlplane/transformation/kms_transformation_test.go
@@ -145,7 +145,7 @@ resources:
 	// Since Data Encryption Key (DEK) is randomly generated (per encryption operation), we need to ask KMS Mock for it.
 	plainTextDEK := pluginMock.LastEncryptRequest()
 
-	secretETCDPath := test.getETCDPath()
+	secretETCDPath := test.getETCDPathForResource("", "secrets", test.secret.Name, test.secret.Namespace)
 	rawEnvelope, err := test.getRawSecretFromETCD()
 	if err != nil {
 		t.Fatalf("failed to read %s from etcd: %v", secretETCDPath, err)
diff --git a/test/integration/controlplane/transformation/kmsv2_transformation_test.go b/test/integration/controlplane/transformation/kmsv2_transformation_test.go
index dfb250a938f..f8b385adb24 100644
--- a/test/integration/controlplane/transformation/kmsv2_transformation_test.go
+++ b/test/integration/controlplane/transformation/kmsv2_transformation_test.go
@@ -154,7 +154,7 @@ resources:
 	// Since Data Encryption Key (DEK) is randomly generated (per encryption operation), we need to ask KMS Mock for it.
 	plainTextDEK := pluginMock.LastEncryptRequest()
 
-	secretETCDPath := test.getETCDPath()
+	secretETCDPath := test.getETCDPathForResource("", "secrets", test.secret.Name, test.secret.Namespace)
 	rawEnvelope, err := test.getRawSecretFromETCD()
 	if err != nil {
 		t.Fatalf("failed to read %s from etcd: %v", secretETCDPath, err)
diff --git a/test/integration/controlplane/transformation/secrets_transformation_test.go b/test/integration/controlplane/transformation/secrets_transformation_test.go
index b899237a937..ccbd86ef0ff 100644
--- a/test/integration/controlplane/transformation/secrets_transformation_test.go
+++ b/test/integration/controlplane/transformation/secrets_transformation_test.go
@@ -95,7 +95,7 @@ func TestSecretsShouldBeTransformed(t *testing.T) {
 		if err != nil {
 			t.Fatalf("Failed to create test secret, error: %v", err)
 		}
-		test.run(tt.unSealFunc, tt.transformerPrefix)
+		test.runResource(test.logger, tt.unSealFunc, tt.transformerPrefix, "", "v1", "secrets", test.secret.Name, test.secret.Namespace)
 		test.cleanUp()
 	}
 }
diff --git a/test/integration/controlplane/transformation/transformation_testcase.go b/test/integration/controlplane/transformation/transformation_testcase.go
index d5ca2b8fcc7..cea83135866 100644
--- a/test/integration/controlplane/transformation/transformation_testcase.go
+++ b/test/integration/controlplane/transformation/transformation_testcase.go
@@ -114,10 +114,6 @@ func (e *transformTest) cleanUp() {
 	e.kubeAPIServer.TearDownFn()
 }
 
-func (e *transformTest) run(unSealSecretFunc unSealSecret, expectedEnvelopePrefix string) {
-	e.runResource(e.logger, unSealSecretFunc, expectedEnvelopePrefix, "", "v1", "secrets", "", "")
-}
-
 func (e *transformTest) runResource(l kubeapiservertesting.Logger, unSealSecretFunc unSealSecret, expectedEnvelopePrefix,
 	group,
 	version,
@@ -212,17 +208,7 @@ func (e *transformTest) benchmark(b *testing.B) {
 	}
 }
 
-func (e *transformTest) getETCDPath() string {
-	return fmt.Sprintf("/%s/secrets/%s/%s", e.storageConfig.Prefix, e.ns.Name, e.secret.Name)
-}
-
 func (e *transformTest) getETCDPathForResource(group, resource, name, namespaceName string) string {
-	if e.secret != nil {
-		group = ""
-		resource = "secrets"
-		name = testSecret
-		namespaceName = testNamespace
-	}
 	groupResource := resource
 	if group != "" {
 		groupResource = fmt.Sprintf("%s/%s", group, resource)
@@ -234,7 +220,7 @@ func (e *transformTest) getETCDPathForResource(group, resource, name, namespaceN
 }
 
 func (e *transformTest) getRawSecretFromETCD() ([]byte, error) {
-	secretETCDPath := e.getETCDPath()
+	secretETCDPath := e.getETCDPathForResource("", "secrets", e.secret.Name, e.ns.Name)
 	etcdResponse, err := e.readRawRecordFromETCD(secretETCDPath)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read %s from etcd: %v", secretETCDPath, err)
```